SignalMessaging/attachments/ApproveContactShareViewController.swift

```diff
@@ -0,0 +1,556 @@
+        fieldsView.autoPinEdge(toSuperviewEdge: .bottom)
+    }
+
+    private func createFieldsView() -> UIView {
+        SwiftAssertIsOnMainThread(#function)
+
+        var rows = [UIView]()
+
+        rows.append(createNameRow())
+
+        for fieldView in fieldViews {
+            rows.append(fieldView)
+        }
+
+        return ContactFieldView(rows: rows, hMargin: hMargin)
+    }
+
+    private let hMargin = CGFloat(16)
+
+    func createNameRow() -> UIView {
+        let nameVMargin = CGFloat(16)
+
+        let row = UIView()
+        row.layoutMargins = UIEdgeInsets(top: nameVMargin, left: 0, bottom: nameVMargin, right: 0)
+
+        let stackView = UIStackView()
+        stackView.axis = .horizontal
+        stackView.alignment = .center
+        stackView.layoutMargins = .zero
+        stackView.spacing = 10
+        row.addSubview(stackView)
+        stackView.autoPinLeadingToSuperviewMargin()
+        stackView.autoPinTrailingToSuperviewMargin()
+        stackView.autoPinTopToSuperviewMargin()
+        stackView.autoPinBottomToSuperviewMargin()
+
+        let nameLabel = UILabel()
+        self.nameLabel = nameLabel
+        nameLabel.text = contactShare.displayName
+        nameLabel.font = UIFont.ows_dynamicTypeBody
+        nameLabel.textColor = UIColor.ows_materialBlue
+        nameLabel.lineBreakMode = .byTruncatingTail
+        stackView.addArrangedSubview(nameLabel)
+        nameLabel.setContentHuggingHigh()
+
+        let editNameLabel = UILabel()
+        editNameLabel.text = NSLocalizedString("CONTACT_EDIT_NAME_BUTTON", comment: "Label for the 'edit name' button in the contact share approval view.")
+        editNameLabel.font = UIFont.ows_dynamicTypeCaption1
+        editNameLabel.textColor = UIColor.black
+        stackView.addArrangedSubview(editNameLabel)
+        editNameLabel.setContentHuggingHigh()
+        editNameLabel.setCompressionResistanceHigh()
+
+        // Icon
+        let iconName = (self.view.isRTL() ? "system_disclosure_indicator_rtl" : "system_disclosure_indicator")
+        guard let iconImage = UIImage(named: iconName) else {
+            owsFail("\(logTag) missing icon.")
+            return row
+        }
+        let iconView = UIImageView(image: iconImage.withRenderingMode(.alwaysTemplate))
+        iconView.contentMode = .scaleAspectFit
+        iconView.tintColor = UIColor.black.withAlphaComponent(0.6)
+        stackView.addArrangedSubview(iconView)
+        iconView.setContentHuggingHigh()
+        iconView.setCompressionResistanceHigh()
+
+        row.isUserInteractionEnabled = true
+        row.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(didPressEditName)))
+
+        return row
+    }
+
+    func previewView(forPhoneNumber phoneNumber: OWSContactPhoneNumber) -> UIView {
+        let label = UILabel()
+        label.text = PhoneNumber.bestEffortLocalizedPhoneNumber(withE164: phoneNumber.phoneNumber)
+        label.font = UIFont.ows_dynamicTypeCaption1
```

Comment:
consider pulling out these font attributes to design constants so we don't have to update them in three+ places.

Reply:
They've been de-duplicated in a subsequent PR.